src/tail/tunnel.rs

```diff
@@ -1,29 +1,27 @@
 use std::path::PathBuf;
 use std::process::Stdio;
 use std::str;
-use std::thread;
-use std::time::Duration;
 
 use log::log_enabled;
 use log::Level::Info;
 use tokio::process::Child;
 use tokio::process::Command;
+use tokio::sync::oneshot::Receiver;
 
 pub struct Tunnel {
     child: Child,
-    command_name: String,
 }
 
+/// Tunnel wraps a child process that runs cloudflared and forwards requests from the Trace Worker
+/// in the runtime to our local LogServer instance. We wrap it in a struct primarily to hold the
+/// state of the child process so that upon receipt of a SIGINT message we can more swiftly kill it
+/// and wait on its output; otherwise we end up leaving it behind when wrangler exits and this
+/// causes problems if it still exists the next time we start up a tail.
```

Comment:
these problems should still be handled, what happens if the user spams ctrl+c and graceful shutdown doesn't happen? should make new issue

Reply:
so fun fact, what we do here is actually the same as sending another ctrl+c... there's really no time for the user to do it more than once.